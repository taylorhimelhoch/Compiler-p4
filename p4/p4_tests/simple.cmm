string glob;
int glob1;
string glob2;
int main(int a, bool b){
	int c;
	glob = a;
	glob1 = @a + &b;
	glob1 = a - c;
	glob1 = glob1 / glob2;
	glob1 = glob1 * glob1;
	glob1 = glob1 and glob2;
	glob2 = a or b;
	glob2 = a != b;
	glob = a <= b;
	glob = a < b;
	glob = a > b;
	glob = a >= b;
	glob = a==b;
}
