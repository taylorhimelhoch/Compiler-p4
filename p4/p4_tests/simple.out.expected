bool glob;
int glob1;
string glob2;
int main(int a, bool b){
    int c;
    glob (bool)  = a;
	glob1 (int) = (a (int) + b (bool));
	glob1 (int) = (a (int) - c (int));
	glob1 (int) = (glob1 (int) / glob2 (string));
	glob1 (int) = (glob1 (int) * glob1 (int));
	glob1 (int) = (glob1 (int) and glob2 (string));
	glob2 (string) = (a (int) or b (bool));
	glob2 (string) = (a (int) != b (bool));
	glob (bool) = (a (int) <= b (bool));
	glob (bool) = (a (int) < b (bool));
	glob (bool) = (a (int) > b (bool));
	glob (bool) = (a (int) >= b (bool));
	glob (bool) = (a (int) == b (bool));
}